torchao/quantization/subclass.py

```diff
@@ -134,14 +139,21 @@ def __torch_function__(cls, func, types, args=(), kwargs=None):
 
     @classmethod
     def __torch_dispatch__(cls, func, types, args, kwargs):
+        # Note: we only added cpu path here for 8da4w, this is for executorch, in the future
+        # 1. we'll add cpu/cuda version (int4mm etc.)
+        # 2. we'll need to hide the 8da4w executorch version under things like layouts (we also have multiple impl for cpu kernel as Michael mentioned), so it will be something like
```

Comment:
This definitely worries me. What is et layout really? And I dont feel we have put enough thoughts on whether hiding it like this, to dispatch to weight only quantization vs. weight + dyn act quantization, is a good idea

Reply:
oh, et layout just means we don't use packed weights + dispatch to int4mm kernels (as what we are doing for cpu and cuda device), since executorch does not have their own device, so we are reusing cpu as the device for executorch path, and thus we need some extra information (layout or something else) to generate the q/dq format the executorch is consuming.

Reply:
we are dispatching on weight only quant v.s. weight + dyn act quant, not sure what you mean by hiding actually, I'm putting up another PR soon, it will it it clearer

Reply:
If you want to do subclass based approach, I would expect each different flavor to inherit perhaps from AffineQuantizedTensor class, e.g.
```
class X86WeightOnlyQuantizedLinearWeightTensor(AffineQuantizedTensor):
    def __init__(...):
         do packing here
    def __torch_dispatch__(....):
          dispatch to x86 impl
```